### ISSUE TITLE
Bug: `VideoBoxes` breaks inherited `Boxes` methods — `get_boxes_shape`/`to_mask` raise `TypeError`, indexing drops `temporal_channel_size`

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4249]
`VideoBoxes` (exported and documented by #4138) overrides `to_tensor(self, mode=None)` without the `as_padded_sequence` keyword that the inherited `Boxes.get_boxes_shape` and `Boxes.to_mask` pass, so both raise `TypeError` on a `VideoBoxes`. Separately, `Boxes.__getitem__` builds the result with `type(self)(...)` and never sets `temporal_channel_size`, so `video_boxes[0].to_tensor()` raises `AttributeError`. Methods that copy through `clone()` (`transform_boxes`, `translate`, `clamp`, `filter_boxes_by_area`, `pad`, `unpad`, `merge`, `index_put`, `to`, `type`) keep the temporal size and work.

Reproduced 2026-09-05 at `1eea5835`, PyTorch 2.14.0, Python 3.11, CPU and MPS.

## Reproduction

```python
import torch
from kornia.geometry.boxes import VideoBoxes

frame = torch.tensor([[[1., 1.], [3., 1.], [3., 3.], [1., 3.]]])  # (1, 4, 2)
video = VideoBoxes.from_tensor(frame.view(1, 1, 1, 4, 2).expand(2, 3, 1, 4, 2).contiguous())  # (B=2, T=3)
for label, call in (
    ("get_boxes_shape", lambda: video.get_boxes_shape()),
    ("to_mask", lambda: video.to_mask(4, 5)),
    ("[0].to_tensor", lambda: video[0].to_tensor()),
    ("clamp keeps T", lambda: video.clamp(torch.zeros(6, 2), torch.full((6, 2), 2.0)).temporal_channel_size),
):
    try:
        print(label, "->", call())
    except Exception as error:
        print(label, "->", type(error).__name__, str(error)[:80])
```

## Observed

```text
get_boxes_shape -> TypeError VideoBoxes.to_tensor() got an unexpected keyword argument 'as_padded_sequence'
to_mask -> TypeError VideoBoxes.to_tensor() got an unexpected keyword argument 'as_padded_sequence'
[0].to_tensor -> AttributeError 'VideoBoxes' object has no attribute 'temporal_channel_size'
clamp keeps T -> 3
```

## Expected

The override should accept the base-class keyword (forwarding or ignoring it) so inherited methods work, and `__getitem__` should carry `temporal_channel_size` (or `VideoBoxes` should override it). Focused fix, welcome as a PR. What `get_boxes_shape` and `to_mask` should return for the temporal wrapper, `(B*T, N)` like `Boxes` or `(B, T, N)` with the axis restored, is the one design question; the batch-4c strict xfail (`test_convention_inherited_shape_and_mask_work_on_the_temporal_wrapper_4249`) asserts only element counts so either answer clears it.

#4138 recorded two related warts on the same class (inert `validate_boxes`, unreachable list branch in `to_tensor`); this issue is the inherited-method breakage it did not cover. `AugmentationSequential` (`kornia/augmentation/container/augment.py:635`) builds `VideoBoxes` and calls `transform_boxes`/`to_tensor` only, so the pipeline path is unaffected.

## Classification

Broken output: valid public-API calls raise `TypeError`/`AttributeError`. Idiom 3 (a subclass narrows the inherited method contract). Two symptoms, one class.

## Disposition

n/a — single symbol

Found during the bbox/boxes conventions audit batch 4c (#4020); the audit changes no runtime behavior.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
